EXCEPTION: Uncaught (in promise): TypeError: Cannot read property 'visitExpression' of undefined

<!--
IF YOU DON'T FILL OUT THE FOLLOWING INFORMATION WE MIGHT CLOSE YOUR ISSUE WITHOUT INVESTIGATING
-->

**I'm submitting a ...**  (check one with "x")

```
[x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**

```
error_handler.js:46 EXCEPTION: Uncaught (in promise): TypeError: Cannot read property 'visitExpression' of undefined
TypeError: Cannot read property 'visitExpression' of undefined
    at http://localhost:4000/vendor.bundle.js:14857:77
    at JitEmitterVisitor.AbstractEmitterVisitor.visitAllObjects (http://localhost:4000/vendor.bundle.js:14865:13)
    at JitEmitterVisitor.AbstractEmitterVisitor.visitAllExpressions (http://localhost:4000/vendor.bundle.js:14857:14)
    at JitEmitterVisitor.AbstractEmitterVisitor.visitLiteralArrayExpr (http://localhost:4000/vendor.bundle.js:14836:14)
    at LiteralArrayExpr.visitExpression (http://localhost:4000/vendor.bundle.js:15581:24)
    at http://localhost:4000/vendor.bundle.js:14857:78
    at JitEmitterVisitor.AbstractEmitterVisitor.visitAllObjects (http://localhost:4000/vendor.bundle.js:14865:13)
    at JitEmitterVisitor.AbstractEmitterVisitor.visitAllExpressions (http://localhost:4000/vendor.bundle.js:14857:14)
    at JitEmitterVisitor.AbstractEmitterVisitor.visitLiteralArrayExpr (http://localhost:4000/vendor.bundle.js:14836:14)
    at LiteralArrayExpr.visitExpression (http://localhost:4000/vendor.bundle.js:15581:24)
    at JitEmitterVisitor.AbstractEmitterVisitor.visitWritePropExpr (http://localhost:4000/vendor.bundle.js:14673:20)
    at WritePropExpr.visitExpression (http://localhost:4000/vendor.bundle.js:15383:24)
    at JitEmitterVisitor.AbstractEmitterVisitor.visitExpressionStmt (http://localhost:4000/vendor.bundle.js:14592:19)
    at ExpressionStatement.visitStatement (http://localhost:4000/vendor.bundle.js:15661:24)
    at http://localhost:4000/vendor.bundle.js:14873:58
    at Array.forEach (native)
    at JitEmitterVisitor.AbstractEmitterVisitor.visitAllStatements (http://localhost:4000/vendor.bundle.js:14873:20)
    at JitEmitterVisitor.AbstractEmitterVisitor.visitIfStmt (http://localhost:4000/vendor.bundle.js:14609:18)
    at IfStmt.visitStatement (http://localhost:4000/vendor.bundle.js:15748:24)
    at http://localhost:4000/vendor.bundle.js:14873:58
    at Array.forEach (native)
    at JitEmitterVisitor.AbstractEmitterVisitor.visitAllStatements (http://localhost:4000/vendor.bundle.js:14873:20)
    at JitEmitterVisitor.AbstractJsEmitterVisitor._visitClassGetter (http://localhost:4000/vendor.bundle.js:14973:18)
    at http://localhost:4000/vendor.bundle.js:14947:63
    at Array.forEach (native)
    at JitEmitterVisitor.AbstractJsEmitterVisitor.visitDeclareClassStmt (http://localhost:4000/vendor.bundle.js:14947:22)
    at ClassStmt.visitStatement (http://localhost:4000/vendor.bundle.js:15734:24)
    at http://localhost:4000/vendor.bundle.js:14873:58
    at Array.forEach (native)
    at JitEmitterVisitor.AbstractEmitterVisitor.visitAllStatements (http://localhost:4000/vendor.bundle.js:14873:20)
    at jitStatements (http://localhost:4000/vendor.bundle.js:16445:15)
    at RuntimeCompiler._compileModule (http://localhost:4000/vendor.bundle.js:17940:131)
    at RuntimeCompiler._compileModuleAndComponents (http://localhost:4000/vendor.bundle.js:17892:36)
    at RuntimeCompiler.compileModuleAsync (http://localhost:4000/vendor.bundle.js:17882:21)
    at ModuleBoundCompiler.compileModuleAsync (http://localhost:4000/vendor.bundle.js:18200:31)
    at MergeMapSubscriber.project (http://localhost:4000/vendor.bundle.js:51935:440)
    at MergeMapSubscriber._tryNext (http://localhost:4000/vendor.bundle.js:87167:27)
    at MergeMapSubscriber._next (http://localhost:4000/vendor.bundle.js:87157:18)
    at MergeMapSubscriber.Subscriber.next (http://localhost:4000/vendor.bundle.js:85209:18)
    at http://localhost:4000/vendor.bundle.js:86202:36
    at ZoneDelegate.invoke (http://localhost:4000/polyfills.bundle.js:15310:26)
    at Object.onInvoke (http://localhost:4000/vendor.bundle.js:36136:37)
    at ZoneDelegate.invoke (http://localhost:4000/polyfills.bundle.js:15309:32)
    at Zone.run (http://localhost:4000/polyfills.bundle.js:15192:43)
    at http://localhost:4000/polyfills.bundle.js:15580:57
    at ZoneDelegate.invokeTask (http://localhost:4000/polyfills.bundle.js:15343:35)
    at Object.onInvokeTask (http://localhost:4000/vendor.bundle.js:36127:37)
    at ZoneDelegate.invokeTask (http://localhost:4000/polyfills.bundle.js:15342:40)
    at Zone.runTask (http://localhost:4000/polyfills.bundle.js:15232:47)
    at drainMicroTaskQueue (http://localhost:4000/polyfills.bundle.js:15479:35)
```

**Expected behavior**
show page correctly.

**Minimal reproduction of the problem with instructions**
http://plnkr.co/edit/pjjwq8jJQHNUG0HK0Z7T?p=info

login component

**What is the motivation / use case for changing the behavior?**

**Please tell us about your environment:**
- **Angular version:** 2.0.0
- **Browser:** [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ]
  
  <!-- All browsers where this could be reproduced -->
- **Language:** [all | TypeScript 2.X | ES6/7 | ES5]
- **Node (for AoT issues):** `node --version` =   5.0.0

I can't provide all of codes, and I use [ng2-Admin](https://github.com/akveo/ng2-admin) panel.

errors:
```
error_handler.js:46EXCEPTION: Uncaught (in promise): TypeError: Cannot read property 'visitExpression' of undefined
TypeError: Cannot read property 'visitExpression' of undefined
    at http://localhost:4000/vendor.bundle.js:14857:77
    at JitEmitterVisitor.AbstractEmitterVisitor.visitAllObjects (http://localhost:4000/vendor.bundle.js:14865:13)
    at JitEmitterVisitor.AbstractEmitterVisitor.visitAllExpressions (http://localhost:4000/vendor.bundle.js:14857:14)
    at JitEmitterVisitor.AbstractEmitterVisitor.visitLiteralArrayExpr (http://localhost:4000/vendor.bundle.js:14836:14)
    at LiteralArrayExpr.visitExpression (http://localhost:4000/vendor.bundle.js:15581:24)
    at http://localhost:4000/vendor.bundle.js:14857:78
    at JitEmitterVisitor.AbstractEmitterVisitor.visitAllObjects (http://localhost:4000/vendor.bundle.js:14865:13)
    at JitEmitterVisitor.AbstractEmitterVisitor.visitAllExpressions (http://localhost:4000/vendor.bundle.js:14857:14)
    at JitEmitterVisitor.AbstractEmitterVisitor.visitLiteralArrayExpr (http://localhost:4000/vendor.bundle.js:14836:14)
    at LiteralArrayExpr.visitExpression (http://localhost:4000/vendor.bundle.js:15581:24)
    at JitEmitterVisitor.AbstractEmitterVisitor.visitWritePropExpr (http://localhost:4000/vendor.bundle.js:14673:20)
    at WritePropExpr.visitExpression (http://localhost:4000/vendor.bundle.js:15383:24)
    at JitEmitterVisitor.AbstractEmitterVisitor.visitExpressionStmt (http://localhost:4000/vendor.bundle.js:14592:19)
    at ExpressionStatement.visitStatement (http://localhost:4000/vendor.bundle.js:15661:24)
    at http://localhost:4000/vendor.bundle.js:14873:58
    at Array.forEach (native)
    at JitEmitterVisitor.AbstractEmitterVisitor.visitAllStatements (http://localhost:4000/vendor.bundle.js:14873:20)
    at JitEmitterVisitor.AbstractEmitterVisitor.visitIfStmt (http://localhost:4000/vendor.bundle.js:14609:18)
    at IfStmt.visitStatement (http://localhost:4000/vendor.bundle.js:15748:24)
    at http://localhost:4000/vendor.bundle.js:14873:58
    at Array.forEach (native)
    at JitEmitterVisitor.AbstractEmitterVisitor.visitAllStatements (http://localhost:4000/vendor.bundle.js:14873:20)
    at JitEmitterVisitor.AbstractJsEmitterVisitor._visitClassGetter (http://localhost:4000/vendor.bundle.js:14973:18)
    at http://localhost:4000/vendor.bundle.js:14947:63
    at Array.forEach (native)
    at JitEmitterVisitor.AbstractJsEmitterVisitor.visitDeclareClassStmt (http://localhost:4000/vendor.bundle.js:14947:22)
    at ClassStmt.visitStatement (http://localhost:4000/vendor.bundle.js:15734:24)
    at http://localhost:4000/vendor.bundle.js:14873:58
    at Array.forEach (native)
    at JitEmitterVisitor.AbstractEmitterVisitor.visitAllStatements (http://localhost:4000/vendor.bundle.js:14873:20)
    at jitStatements (http://localhost:4000/vendor.bundle.js:16445:15)
    at RuntimeCompiler._compileModule (http://localhost:4000/vendor.bundle.js:17940:131)
    at RuntimeCompiler._compileModuleAndComponents (http://localhost:4000/vendor.bundle.js:17892:36)
    at RuntimeCompiler.compileModuleAsync (http://localhost:4000/vendor.bundle.js:17882:21)
    at ModuleBoundCompiler.compileModuleAsync (http://localhost:4000/vendor.bundle.js:18200:31)
    at MergeMapSubscriber.project (http://localhost:4000/vendor.bundle.js:51935:440)
    at MergeMapSubscriber._tryNext (http://localhost:4000/vendor.bundle.js:87167:27)
    at MergeMapSubscriber._next (http://localhost:4000/vendor.bundle.js:87157:18)
    at MergeMapSubscriber.Subscriber.next (http://localhost:4000/vendor.bundle.js:85209:18)
    at http://localhost:4000/vendor.bundle.js:86202:36
    at ZoneDelegate.invoke (http://localhost:4000/polyfills.bundle.js:15310:26)
    at Object.onInvoke (http://localhost:4000/vendor.bundle.js:36136:37)
    at ZoneDelegate.invoke (http://localhost:4000/polyfills.bundle.js:15309:32)
    at Zone.run (http://localhost:4000/polyfills.bundle.js:15192:43)
    at http://localhost:4000/polyfills.bundle.js:15580:57
    at ZoneDelegate.invokeTask (http://localhost:4000/polyfills.bundle.js:15343:35)
    at Object.onInvokeTask (http://localhost:4000/vendor.bundle.js:36127:37)
    at ZoneDelegate.invokeTask (http://localhost:4000/polyfills.bundle.js:15342:40)
    at Zone.runTask (http://localhost:4000/polyfills.bundle.js:15232:47)
    at drainMicroTaskQueue (http://localhost:4000/polyfills.bundle.js:15479:35)ErrorHandler.handleError @ error_handler.js:46
error_handler.js:51ORIGINAL STACKTRACE:ErrorHandler.handleError @ error_handler.js:51
error_handler.js:52Error: Uncaught (in promise): TypeError: Cannot read property 'visitExpression' of undefined
TypeError: Cannot read property 'visitExpression' of undefined
    at http://localhost:4000/vendor.bundle.js:14857:77
    at JitEmitterVisitor.AbstractEmitterVisitor.visitAllObjects (http://localhost:4000/vendor.bundle.js:14865:13)
    at JitEmitterVisitor.AbstractEmitterVisitor.visitAllExpressions (http://localhost:4000/vendor.bundle.js:14857:14)
    at JitEmitterVisitor.AbstractEmitterVisitor.visitLiteralArrayExpr (http://localhost:4000/vendor.bundle.js:14836:14)
    at LiteralArrayExpr.visitExpression (http://localhost:4000/vendor.bundle.js:15581:24)
    at http://localhost:4000/vendor.bundle.js:14857:78
    at JitEmitterVisitor.AbstractEmitterVisitor.visitAllObjects (http://localhost:4000/vendor.bundle.js:14865:13)
    at JitEmitterVisitor.AbstractEmitterVisitor.visitAllExpressions (http://localhost:4000/vendor.bundle.js:14857:14)
    at JitEmitterVisitor.AbstractEmitterVisitor.visitLiteralArrayExpr (http://localhost:4000/vendor.bundle.js:14836:14)
    at LiteralArrayExpr.visitExpression (http://localhost:4000/vendor.bundle.js:15581:24)
    at JitEmitterVisitor.AbstractEmitterVisitor.visitWritePropExpr (http://localhost:4000/vendor.bundle.js:14673:20)
    at WritePropExpr.visitExpression (http://localhost:4000/vendor.bundle.js:15383:24)
    at JitEmitterVisitor.AbstractEmitterVisitor.visitExpressionStmt (http://localhost:4000/vendor.bundle.js:14592:19)
    at ExpressionStatement.visitStatement (http://localhost:4000/vendor.bundle.js:15661:24)
    at http://localhost:4000/vendor.bundle.js:14873:58
    at Array.forEach (native)
    at JitEmitterVisitor.AbstractEmitterVisitor.visitAllStatements (http://localhost:4000/vendor.bundle.js:14873:20)
    at JitEmitterVisitor.AbstractEmitterVisitor.visitIfStmt (http://localhost:4000/vendor.bundle.js:14609:18)
    at IfStmt.visitStatement (http://localhost:4000/vendor.bundle.js:15748:24)
    at http://localhost:4000/vendor.bundle.js:14873:58
    at Array.forEach (native)
    at JitEmitterVisitor.AbstractEmitterVisitor.visitAllStatements (http://localhost:4000/vendor.bundle.js:14873:20)
    at JitEmitterVisitor.AbstractJsEmitterVisitor._visitClassGetter (http://localhost:4000/vendor.bundle.js:14973:18)
    at http://localhost:4000/vendor.bundle.js:14947:63
    at Array.forEach (native)
    at JitEmitterVisitor.AbstractJsEmitterVisitor.visitDeclareClassStmt (http://localhost:4000/vendor.bundle.js:14947:22)
    at ClassStmt.visitStatement (http://localhost:4000/vendor.bundle.js:15734:24)
    at http://localhost:4000/vendor.bundle.js:14873:58
    at Array.forEach (native)
    at JitEmitterVisitor.AbstractEmitterVisitor.visitAllStatements (http://localhost:4000/vendor.bundle.js:14873:20)
    at jitStatements (http://localhost:4000/vendor.bundle.js:16445:15)
    at RuntimeCompiler._compileModule (http://localhost:4000/vendor.bundle.js:17940:131)
    at RuntimeCompiler._compileModuleAndComponents (http://localhost:4000/vendor.bundle.js:17892:36)
    at RuntimeCompiler.compileModuleAsync (http://localhost:4000/vendor.bundle.js:17882:21)
    at ModuleBoundCompiler.compileModuleAsync (http://localhost:4000/vendor.bundle.js:18200:31)
    at MergeMapSubscriber.project (http://localhost:4000/vendor.bundle.js:51935:440)
    at MergeMapSubscriber._tryNext (http://localhost:4000/vendor.bundle.js:87167:27)
    at MergeMapSubscriber._next (http://localhost:4000/vendor.bundle.js:87157:18)
    at MergeMapSubscriber.Subscriber.next (http://localhost:4000/vendor.bundle.js:85209:18)
    at http://localhost:4000/vendor.bundle.js:86202:36
    at ZoneDelegate.invoke (http://localhost:4000/polyfills.bundle.js:15310:26)
    at Object.onInvoke (http://localhost:4000/vendor.bundle.js:36136:37)
    at ZoneDelegate.invoke (http://localhost:4000/polyfills.bundle.js:15309:32)
    at Zone.run (http://localhost:4000/polyfills.bundle.js:15192:43)
    at http://localhost:4000/polyfills.bundle.js:15580:57
    at ZoneDelegate.invokeTask (http://localhost:4000/polyfills.bundle.js:15343:35)
    at Object.onInvokeTask (http://localhost:4000/vendor.bundle.js:36127:37)
    at ZoneDelegate.invokeTask (http://localhost:4000/polyfills.bundle.js:15342:40)
    at Zone.runTask (http://localhost:4000/polyfills.bundle.js:15232:47)
    at drainMicroTaskQueue (http://localhost:4000/polyfills.bundle.js:15479:35)
    at resolvePromise (http://localhost:4000/polyfills.bundle.js:15546:31)
    at http://localhost:4000/polyfills.bundle.js:15523:13
    at ZoneDelegate.invoke (http://localhost:4000/polyfills.bundle.js:15310:26)
    at Object.onInvoke (http://localhost:4000/vendor.bundle.js:36136:37)
    at ZoneDelegate.invoke (http://localhost:4000/polyfills.bundle.js:15309:32)
    at Zone.run (http://localhost:4000/polyfills.bundle.js:15192:43)
    at http://localhost:4000/polyfills.bundle.js:15580:57
    at ZoneDelegate.invokeTask (http://localhost:4000/polyfills.bundle.js:15343:35)
    at Object.onInvokeTask (http://localhost:4000/vendor.bundle.js:36127:37)
    at ZoneDelegate.invokeTask (http://localhost:4000/polyfills.bundle.js:15342:40)
    at Zone.runTask (http://localhost:4000/polyfills.bundle.js:15232:47)
    at drainMicroTaskQueue (http://localhost:4000/polyfills.bundle.js:15479:35)
  -------------   Elapsed: 2 ms; At: Wed Oct 19 2016 16:00:58 GMT+0330 (Iran Standard Time)   -------------  
    at Object.onScheduleTask (http://localhost:4000/polyfills.bundle.js:14997:18)
    at ZoneDelegate.scheduleTask (http://localhost:4000/polyfills.bundle.js:15320:49)
    at Zone.scheduleMicroTask (http://localhost:4000/polyfills.bundle.js:15246:39)
    at scheduleResolveOrReject (http://localhost:4000/polyfills.bundle.js:15578:14)
    at resolvePromise (http://localhost:4000/polyfills.bundle.js:15541:21)
    at http://localhost:4000/polyfills.bundle.js:15523:13
    at http://localhost:4000/vendor.bundle.js:51514:25
    at ZoneDelegate.invoke (http://localhost:4000/polyfills.bundle.js:15310:26)
    at Object.onInvoke (http://localhost:4000/vendor.bundle.js:36136:37)
    at ZoneDelegate.invoke (http://localhost:4000/polyfills.bundle.js:15309:32)
    at Zone.run (http://localhost:4000/polyfills.bundle.js:15192:43)
    at http://localhost:4000/polyfills.bundle.js:15580:57
    at ZoneDelegate.invokeTask (http://localhost:4000/polyfills.bundle.js:15343:35)
    at Object.onInvokeTask (http://localhost:4000/vendor.bundle.js:36127:37)
    at ZoneDelegate.invokeTask (http://localhost:4000/polyfills.bundle.js:15342:40)
    at Zone.runTask (http://localhost:4000/polyfills.bundle.js:15232:47)
    at drainMicroTaskQueue (http://localhost:4000/polyfills.bundle.js:15479:35)
  -------------   Elapsed: 23 ms; At: Wed Oct 19 2016 16:00:58 GMT+0330 (Iran Standard Time)   -------------  
    at Object.onScheduleTask (http://localhost:4000/polyfills.bundle.js:14997:18)
    at ZoneDelegate.scheduleTask (http://localhost:4000/polyfills.bundle.js:15320:49)
    at Zone.scheduleMicroTask (http://localhost:4000/polyfills.bundle.js:15246:39)
    at scheduleResolveOrReject (http://localhost:4000/polyfills.bundle.js:15578:14)
    at resolvePromise (http://localhost:4000/polyfills.bundle.js:15541:21)
    at SafeSubscriber._error (http://localhost:4000/polyfills.bundle.js:15523:13)
    at SafeSubscriber.__tryOrUnsub (http://localhost:4000/vendor.bundle.js:85343:16)
    at SafeSubscriber.error (http://localhost:4000/vendor.bundle.js:85304:26)
    at Subscriber._error (http://localhost:4000/vendor.bundle.js:85248:26)
    at Subscriber.error (http://localhost:4000/vendor.bundle.js:85222:18)
    at MergeMapSubscriber.Subscriber._error (http://localhost:4000/vendor.bundle.js:85248:26)
    at MergeMapSubscriber.Subscriber.error (http://localhost:4000/vendor.bundle.js:85222:18)
    at MergeMapSubscriber.Subscriber._error (http://localhost:4000/vendor.bundle.js:85248:26)
    at MergeMapSubscriber.Subscriber.error (http://localhost:4000/vendor.bundle.js:85222:18)
    at MapSubscriber.Subscriber._error (http://localhost:4000/vendor.bundle.js:85248:26)
    at MapSubscriber.Subscriber.error (http://localhost:4000/vendor.bundle.js:85222:18)
    at MapSubscriber.Subscriber._error (http://localhost:4000/vendor.bundle.js:85248:26)
    at MapSubscriber.Subscriber.error (http://localhost:4000/vendor.bundle.js:85222:18)
    at MapSubscriber.Subscriber._error (http://localhost:4000/vendor.bundle.js:85248:26)
    at MapSubscriber.Subscriber.error (http://localhost:4000/vendor.bundle.js:85222:18)
    at MergeMapSubscriber.Subscriber._error (http://localhost:4000/vendor.bundle.js:85248:26)
    at MergeMapSubscriber.Subscriber.error (http://localhost:4000/vendor.bundle.js:85222:18)
    at CatchSubscriber.error (http://localhost:4000/vendor.bundle.js:86417:34)
    at MapSubscriber.Subscriber._error (http://localhost:4000/vendor.bundle.js:85248:26)
    at MapSubscriber.Subscriber.error (http://localhost:4000/vendor.bundle.js:85222:18)
    at MapSubscriber.Subscriber._error (http://localhost:4000/vendor.bundle.js:85248:26)
    at MapSubscriber.Subscriber.error (http://localhost:4000/vendor.bundle.js:85222:18)
    at MapSubscriber.Subscriber._error (http://localhost:4000/vendor.bundle.js:85248:26)
    at MapSubscriber.Subscriber.error (http://localhost:4000/vendor.bundle.js:85222:18)
    at LastSubscriber.Subscriber._error (http://localhost:4000/vendor.bundle.js:85248:26)
    at LastSubscriber.Subscriber.error (http://localhost:4000/vendor.bundle.js:85222:18)
    at MergeAllSubscriber.OuterSubscriber.notifyError (http://localhost:4000/vendor.bundle.js:84891:26)
    at InnerSubscriber._error (http://localhost:4000/vendor.bundle.js:84556:21)
    at InnerSubscriber.Subscriber.error (http://localhost:4000/vendor.bundle.js:85222:18)
    at MapSubscriber.Subscriber._error (http://localhost:4000/vendor.bundle.js:85248:26)
    at MapSubscriber.Subscriber.error (http://localhost:4000/vendor.bundle.js:85222:18)
    at CatchSubscriber.error (http://localhost:4000/vendor.bundle.js:86417:34)
    at FirstSubscriber.Subscriber._error (http://localhost:4000/vendor.bundle.js:85248:26)
    at FirstSubscriber.Subscriber.error (http://localhost:4000/vendor.bundle.js:85222:18)
    at MergeAllSubscriber.OuterSubscriber.notifyError (http://localhost:4000/vendor.bundle.js:84891:26)
    at InnerSubscriber._error (http://localhost:4000/vendor.bundle.js:84556:21)
    at InnerSubscriber.Subscriber.error (http://localhost:4000/vendor.bundle.js:85222:18)
    at CatchSubscriber.error (http://localhost:4000/vendor.bundle.js:86417:34)
    at MergeMapSubscriber.Subscriber._error (http://localhost:4000/vendor.bundle.js:85248:26)
    at MergeMapSubscriber.Subscriber.error (http://localhost:4000/vendor.bundle.js:85222:18)
    at MergeMapSubscriber.OuterSubscriber.notifyError (http://localhost:4000/vendor.bundle.js:84891:26)
    at InnerSubscriber._error (http://localhost:4000/vendor.bundle.js:84556:21)
    at InnerSubscriber.Subscriber.error (http://localhost:4000/vendor.bundle.js:85222:18)
    at MapSubscriber.Subscriber._error (http://localhost:4000/vendor.bundle.js:85248:26)
    at MapSubscriber.Subscriber.error (http://localhost:4000/vendor.bundle.js:85222:18)
    at MapSubscriber.Subscriber._error (http://localhost:4000/vendor.bundle.js:85248:26)
    at MapSubscriber.Subscriber.error (http://localhost:4000/vendor.bundle.js:85222:18)
    at MergeMapSubscriber._tryNext (http://localhost:4000/vendor.bundle.js:87170:30)
    at MergeMapSubscriber._next (http://localhost:4000/vendor.bundle.js:87157:18)
    at MergeMapSubscriber.Subscriber.next (http://localhost:4000/vendor.bundle.js:85209:18)
    at http://localhost:4000/vendor.bundle.js:86202:36
    at ZoneDelegate.invoke (http://localhost:4000/polyfills.bundle.js:15310:26)
    at Object.onInvoke (http://localhost:4000/vendor.bundle.js:36136:37)
    at ZoneDelegate.invoke (http://localhost:4000/polyfills.bundle.js:15309:32)
    at Zone.run (http://localhost:4000/polyfills.bundle.js:15192:43)
    at http://localhost:4000/polyfills.bundle.js:15580:57
    at ZoneDelegate.invokeTask (http://localhost:4000/polyfills.bundle.js:15343:35)
    at Object.onInvokeTask (http://localhost:4000/vendor.bundle.js:36127:37)
    at ZoneDelegate.invokeTask (http://localhost:4000/polyfills.bundle.js:15342:40)
    at Zone.runTask (http://localhost:4000/polyfills.bundle.js:15232:47)
    at drainMicroTaskQueue (http://localhost:4000/polyfills.bundle.js:15479:35)
  -------------   Elapsed: 6 ms; At: Wed Oct 19 2016 16:00:58 GMT+0330 (Iran Standard Time)   -------------  
    at Object.onScheduleTask (http://localhost:4000/polyfills.bundle.js:14997:18)
    at ZoneDelegate.scheduleTask (http://localhost:4000/polyfills.bundle.js:15320:49)
    at Zone.scheduleMicroTask (http://localhost:4000/polyfills.bundle.js:15246:39)
    at scheduleResolveOrReject (http://localhost:4000/polyfills.bundle.js:15578:14)
    at resolvePromise (http://localhost:4000/polyfills.bundle.js:15541:21)
    at http://localhost:4000/polyfills.bundle.js:15580:17
    at ZoneDelegate.invokeTask (http://localhost:4000/polyfills.bundle.js:15343:35)
    at Object.onInvokeTask (http://localhost:4000/vendor.bundle.js:36127:37)
    at ZoneDelegate.invokeTask (http://localhost:4000/polyfills.bundle.js:15342:40)
    at Zone.runTask (http://localhost:4000/polyfills.bundle.js:15232:47)
    at drainMicroTaskQueue (http://localhost:4000/polyfills.bundle.js:15479:35)
  -------------   Elapsed: 0 ms; At: Wed Oct 19 2016 16:00:58 GMT+0330 (Iran Standard Time)   -------------  
    at Object.onScheduleTask (http://localhost:4000/polyfills.bundle.js:14997:18)
    at ZoneDelegate.scheduleTask (http://localhost:4000/polyfills.bundle.js:15320:49)
    at Zone.scheduleMicroTask (http://localhost:4000/polyfills.bundle.js:15246:39)
    at scheduleResolveOrReject (http://localhost:4000/polyfills.bundle.js:15578:14)
    at resolvePromise (http://localhost:4000/polyfills.bundle.js:15541:21)
    at http://localhost:4000/polyfills.bundle.js:15580:17
    at ZoneDelegate.invokeTask (http://localhost:4000/polyfills.bundle.js:15343:35)
    at Object.onInvokeTask (http://localhost:4000/vendor.bundle.js:36127:37)
    at ZoneDelegate.invokeTask (http://localhost:4000/polyfills.bundle.js:15342:40)
    at Zone.runTask (http://localhost:4000/polyfills.bundle.js:15232:47)
    at drainMicroTaskQueue (http://localhost:4000/polyfills.bundle.js:15479:35)
  -------------   Elapsed: 39 ms; At: Wed Oct 19 2016 16:00:58 GMT+0330 (Iran Standard Time)   -------------  
    at Object.onScheduleTask (http://localhost:4000/polyfills.bundle.js:14997:18)
    at ZoneDelegate.scheduleTask (http://localhost:4000/polyfills.bundle.js:15320:49)
    at Zone.scheduleMicroTask (http://localhost:4000/polyfills.bundle.js:15246:39)
    at scheduleResolveOrReject (http://localhost:4000/polyfills.bundle.js:15578:14)
    at resolvePromise (http://localhost:4000/polyfills.bundle.js:15541:21)
    at http://localhost:4000/polyfills.bundle.js:15523:13
    at webpackJsonpCallback (http://localhost:4000/polyfills.bundle.js:22:29)
    at http://localhost:4000/4.chunk.js:1:1ErrorHandler.handleError @ error_handler.js:52
zone.js:388Unhandled Promise rejection: Cannot read property 'visitExpression' of undefined ; Zone: angular ; Task: Promise.then ; Value: TypeError: Cannot read property 'visitExpression' of undefined(…) TypeError: Cannot read property 'visitExpression' of undefined
    at http://localhost:4000/vendor.bundle.js:14857:77
    at JitEmitterVisitor.AbstractEmitterVisitor.visitAllObjects (http://localhost:4000/vendor.bundle.js:14865:13)
    at JitEmitterVisitor.AbstractEmitterVisitor.visitAllExpressions (http://localhost:4000/vendor.bundle.js:14857:14)
    at JitEmitterVisitor.AbstractEmitterVisitor.visitLiteralArrayExpr (http://localhost:4000/vendor.bundle.js:14836:14)
    at LiteralArrayExpr.visitExpression (http://localhost:4000/vendor.bundle.js:15581:24)
    at http://localhost:4000/vendor.bundle.js:14857:78
    at JitEmitterVisitor.AbstractEmitterVisitor.visitAllObjects (http://localhost:4000/vendor.bundle.js:14865:13)
    at JitEmitterVisitor.AbstractEmitterVisitor.visitAllExpressions (http://localhost:4000/vendor.bundle.js:14857:14)
    at JitEmitterVisitor.AbstractEmitterVisitor.visitLiteralArrayExpr (http://localhost:4000/vendor.bundle.js:14836:14)
    at LiteralArrayExpr.visitExpression (http://localhost:4000/vendor.bundle.js:15581:24)
    at JitEmitterVisitor.AbstractEmitterVisitor.visitWritePropExpr (http://localhost:4000/vendor.bundle.js:14673:20)
    at WritePropExpr.visitExpression (http://localhost:4000/vendor.bundle.js:15383:24)
    at JitEmitterVisitor.AbstractEmitterVisitor.visitExpressionStmt (http://localhost:4000/vendor.bundle.js:14592:19)
    at ExpressionStatement.visitStatement (http://localhost:4000/vendor.bundle.js:15661:24)
    at http://localhost:4000/vendor.bundle.js:14873:58
    at Array.forEach (native)
    at JitEmitterVisitor.AbstractEmitterVisitor.visitAllStatements (http://localhost:4000/vendor.bundle.js:14873:20)
    at JitEmitterVisitor.AbstractEmitterVisitor.visitIfStmt (http://localhost:4000/vendor.bundle.js:14609:18)
    at IfStmt.visitStatement (http://localhost:4000/vendor.bundle.js:15748:24)
    at http://localhost:4000/vendor.bundle.js:14873:58
    at Array.forEach (native)
    at JitEmitterVisitor.AbstractEmitterVisitor.visitAllStatements (http://localhost:4000/vendor.bundle.js:14873:20)
    at JitEmitterVisitor.AbstractJsEmitterVisitor._visitClassGetter (http://localhost:4000/vendor.bundle.js:14973:18)
    at http://localhost:4000/vendor.bundle.js:14947:63
    at Array.forEach (native)
    at JitEmitterVisitor.AbstractJsEmitterVisitor.visitDeclareClassStmt (http://localhost:4000/vendor.bundle.js:14947:22)
    at ClassStmt.visitStatement (http://localhost:4000/vendor.bundle.js:15734:24)
    at http://localhost:4000/vendor.bundle.js:14873:58
    at Array.forEach (native)
    at JitEmitterVisitor.AbstractEmitterVisitor.visitAllStatements (http://localhost:4000/vendor.bundle.js:14873:20)
    at jitStatements (http://localhost:4000/vendor.bundle.js:16445:15)
    at RuntimeCompiler._compileModule (http://localhost:4000/vendor.bundle.js:17940:131)
    at RuntimeCompiler._compileModuleAndComponents (http://localhost:4000/vendor.bundle.js:17892:36)
    at RuntimeCompiler.compileModuleAsync (http://localhost:4000/vendor.bundle.js:17882:21)
    at ModuleBoundCompiler.compileModuleAsync (http://localhost:4000/vendor.bundle.js:18200:31)
    at MergeMapSubscriber.project (http://localhost:4000/vendor.bundle.js:51935:440)
    at MergeMapSubscriber._tryNext (http://localhost:4000/vendor.bundle.js:87167:27)
    at MergeMapSubscriber._next (http://localhost:4000/vendor.bundle.js:87157:18)
    at MergeMapSubscriber.Subscriber.next (http://localhost:4000/vendor.bundle.js:85209:18)
    at http://localhost:4000/vendor.bundle.js:86202:36
    at ZoneDelegate.invoke (http://localhost:4000/polyfills.bundle.js:15310:26)
    at Object.onInvoke (http://localhost:4000/vendor.bundle.js:36136:37)
    at ZoneDelegate.invoke (http://localhost:4000/polyfills.bundle.js:15309:32)
    at Zone.run (http://localhost:4000/polyfills.bundle.js:15192:43)
    at http://localhost:4000/polyfills.bundle.js:15580:57
    at ZoneDelegate.invokeTask (http://localhost:4000/polyfills.bundle.js:15343:35)
    at Object.onInvokeTask (http://localhost:4000/vendor.bundle.js:36127:37)
    at ZoneDelegate.invokeTask (http://localhost:4000/polyfills.bundle.js:15342:40)
    at Zone.runTask (http://localhost:4000/polyfills.bundle.js:15232:47)
    at drainMicroTaskQueue (http://localhost:4000/polyfills.bundle.js:15479:35)consoleError @ zone.js:388
zone.js:390Error: Uncaught (in promise): TypeError: Cannot read property 'visitExpression' of undefined(…)consoleError @ zone.js:390

```

--- a/modules/@angular/router/src/config.ts
+++ b/modules/@angular/router/src/config.ts
@@ -313,10 +313,26 @@ export interface Route {
 }
 
 export function validateConfig(config: Routes): void {
-  config.forEach(validateNode);
+  // forEach doesn't iterate undefined values
+  for (let i = 0; i < config.length; i++) {
+    validateNode(config[i]);
+  }
 }
 
 function validateNode(route: Route): void {
+  if (!route) {
+    throw new Error(`
+      Invalid route configuration: Encountered undefined route.
+      The reason might be an extra comma.
+       
+      Example: 
+      const routes: Routes = [
+        { path: '', redirectTo: '/dashboard', pathMatch: 'full' },
+        { path: 'dashboard',  component: DashboardComponent },, << two commas
+        { path: 'detail/:id', component: HeroDetailComponent }
+      ];
+    `);
+  }
   if (Array.isArray(route)) {
     throw new Error(`Invalid route configuration: Array cannot be specified`);
   }

--- a/modules/@angular/router/src/router_module.ts
+++ b/modules/@angular/router/src/router_module.ts
@@ -8,7 +8,6 @@
 
 import {APP_BASE_HREF, HashLocationStrategy, Location, LocationStrategy, PathLocationStrategy, PlatformLocation} from '@angular/common';
 import {ANALYZE_FOR_ENTRY_COMPONENTS, APP_BOOTSTRAP_LISTENER, ApplicationRef, Compiler, Inject, Injector, ModuleWithProviders, NgModule, NgModuleFactoryLoader, OpaqueToken, Optional, Provider, SkipSelf, SystemJsNgModuleLoader} from '@angular/core';
-
 import {Route, Routes} from './config';
 import {RouterLink, RouterLinkWithHref} from './directives/router_link';
 import {RouterLinkActive} from './directives/router_link_active';

--- a/modules/@angular/router/test/config.spec.ts
+++ b/modules/@angular/router/test/config.spec.ts
@@ -15,6 +15,12 @@ describe('config', () => {
       validateConfig([{path: 'a', redirectTo: 'b'}, {path: 'b', component: ComponentA}]);
     });
 
+    it('should throw for undefined route', () => {
+      expect(() => {
+        validateConfig([{path: 'a', component: ComponentA}, , {path: 'b', component: ComponentB}]);
+      }).toThrowError();
+    });
+
     it('should throw when Array is passed', () => {
       expect(() => {
         validateConfig([

